src/loaders.rs

```diff
@@ -239,6 +239,104 @@ mod tests {
         })
     }
 
+    #[test]
+    fn test_cached_loader() {
+        pyo3::prepare_freethreaded_python();
+
+        Python::with_gil(|py| {
+            // Helper to check cache contents
+            let verify_cache = |cache: &HashMap<String, Result<Template, LoaderError>>,
+                                key: &str,
+                                expected_path: &Path| {
+                if let Some(Ok(cached_template)) = cache.get(key) {
+                    assert_eq!(cached_template.filename.as_ref().unwrap(), expected_path);
+                } else {
+                    panic!("Expected '{}' to be in cache.", key);
+                }
+            };
+
+            // Create a FileSystemLoader for the CachedLoader
+            let filesystem_loader =
+                FileSystemLoader::new(vec!["tests/templates".to_string()], encoding_rs::UTF_8);
+
+            // Wrap the FileSystemLoader in a CachedLoader
+            let mut cached_loader = CachedLoader::new(vec![Loader::FileSystem(filesystem_loader)]);
+
+            // Load a template via the CachedLoader
+            let template = cached_loader
+                .get_template(py, "basic.txt")
+                .expect("Failed to load template")
+                .expect("Template is None");
```

Comment:
Currently the only way this second `Result` can be an error is if the template file has the wrong encoding, so I think this is more accurate:

```suggestion
                .expect("Template file could not be read");
```